### PR TITLE
[SPARK-1442][SQL][follow-up] Address minor comments in Window Function PR (#5604).

### PR DESCRIPTION
**Pull request by yhuai**
Address @marmbrus and @scwf's comments in #5604.

### CODE REVIEW

**Comment by rxin** [comment]
Can you put "window function" on the title also?

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala [hunk 1]
```diff
@@ -638,11 +638,10 @@ class Analyzer(
     def addWindow(windowExpressions: Seq[NamedExpression], child: LogicalPlan): LogicalPlan = {
       // First, we group window expressions based on their Window Spec.
       val groupedWindowExpression = windowExpressions.groupBy { expr =>
-        val windowExpression = expr.find {
-          case window: WindowExpression => true
-          case other => false
-        }.map(_.asInstanceOf[WindowExpression].windowSpec)
-        windowExpression.getOrElse(
+        val windowSpec = expr.collect {
```
Comment (rxin) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala]: collectFirst

**Comment by shivaram** [comment]
@yhuai Could you also check the build error I see in https://github.com/apache/spark/pull/5604#issuecomment-99610082 ?

**Comment by marmbrus** [comment]
Thanks, merged to master and 1.4.

**Pull request closed without merging**